WordleGame/WordleGame/GameViewController.swift

```diff
@@ -12,17 +12,51 @@ class GameViewController: UIViewController {
     @IBOutlet weak var keyboardContainer: KeyboardView!
     @IBOutlet weak var letterContainer: GameFieldView!
     
-    private var gameManager = GameManager()
+    private var gameManager = GameManager() {
+        didSet {
+            gameManager.delegate = self
+        }
+    }
     private let keyboardManager = KeyboardManager()
     
     override func viewDidLoad() {
         super.viewDidLoad()
         
+        addSettingBarItem()
+        
         letterContainer.updateLetterBoxSymbols(gameManager.gameField)
         
+        gameManager.delegate = self
+        
         keyboardContainer.delegate = self
         keyboardContainer.updateKeyboardSymbols(keyboardManager.keyboardSymbols)
     }
+    
+    func addSettingBarItem() {
+        let settingsBarButtonItem = UIBarButtonItem(
+            title: NSString(string: "\u{2699}\u{0000FE0E}") as String,
+            style: .done,
+            target: self,
+            action: #selector(openSettings))
+        let font = UIFont.systemFont(ofSize: 28)
+        let attributes = [NSAttributedString.Key.font : font]
+        settingsBarButtonItem.setTitleTextAttributes(attributes, for: .normal)
+        self.navigationItem.rightBarButtonItem  = settingsBarButtonItem
+    }
+    
+    @objc func openSettings(){
+        guard let settingsViewController = UIStoryboard(name: "Main", bundle: nil)
+                .instantiateViewController(withIdentifier: "settingsVC") as? SettingsViewController else { return }
+        
+        navigationController?.pushViewController(settingsViewController, animated: true)
+    }
+    
+    func restartGame(userName: String) {
+        gameManager.saveData(userName: userName)
```

Comment:
Do you want to save data only on restart? I believe it should be saved when you go back go the menu as well.

Do you want to save data when user is lost as well?

Reply:
Resolved.)